File: alist-web/src/types/automation.ts
export interface AutomationSchedule {
  mode: "interval" | "weekly" | "once"
  interval_value?: number
  interval_unit?: "second" | "minute" | "hour" | "day"
  weekly_day?: string
  time_of_day?: string
  once_at?: string
}

export interface AutomationStepOptions {
  new_name?: string
  inner_path?: string
  password?: string
  put_into_new_dir?: boolean
  cache_full?: boolean
}

export interface AutomationStep {
  action: string
  source: string
  target?: string
  description?: string
  options: AutomationStepOptions
}

export interface AutomationHistory {
  id: number
  success: boolean
  message: string
  started_at: string
  finished_at?: string | null
}

export interface AutomationJob {
  id: number
  name: string
  enabled: boolean
  schedule: AutomationSchedule
  next_run_at?: string | null
  last_run_at?: string | null
  creator_id: number
  steps: AutomationStep[]
  histories: AutomationHistory[]
}

export interface AutomationJobPayload {
  id?: number
  name: string
  enabled: boolean
  schedule: AutomationSchedule
  steps: AutomationStep[]
}

File: alist-backend/internal/model/automation.go
package model

import "time"

type AutomationJob struct {
	ID   uint   `gorm:"primaryKey" json:"id"`
	Name string `json:"name"
                                     gorm:"size:255"`
	Enabled   bool                `json:"enabled"`
	Schedule  string              `gorm:"type:text" json:"schedule"`
	NextRunAt *time.Time          `json:"next_run_at"`
	LastRunAt *time.Time          `json:"last_run_at"`
	CreatorID uint                `json:"creator_id"`
	Steps     []AutomationStep    `json:"steps"`
	Histories []AutomationHistory `json:"histories"`
	CreatedAt time.Time           `json:"created_at"`
	UpdatedAt time.Time           `json:"updated_at"`
}

type AutomationStep struct {
	ID        uint   `gorm:"primaryKey" json:"id"`
	JobID     uint   `json:"job_id"`
	SortOrder int    `json:"sort_order"`
	Action    string `json:"action"`
	Source    string `json:"source"`
	Target    string `json:"target"`
	Options   string `gorm:"type:text" json:"options"`
}

type AutomationHistory struct {
	ID         uint       `gorm:"primaryKey" json:"id"`
	JobID      uint       `json:"job_id"`
	Success    bool       `json:"success"`
	Message    string     `gorm:"type:text" json:"message"`
	StartedAt  time.Time  `json:"started_at"`
	FinishedAt *time.Time `json:"finished_at"`
}
